stubs/toml/toml/encoder.pyi

```diff
@@ -0,0 +1,28 @@
+from _typeshed import SupportsWrite
+from typing import Any, Callable, Iterable, Mapping, MutableMapping
+
+def dump(o: Mapping[str, Any], f: SupportsWrite[str], encoder: TomlEncoder | None = ...) -> str: ...
+def dumps(o: Mapping[str, Any], encoder: TomlEncoder | None = ...) -> str: ...
+
+class TomlEncoder:
+    _dict: type[Mapping[str, Any]]
+    preserve: bool
+    dump_funcs: MutableMapping[type[Any], Callable[[Any], str]]
+    def __init__(self, _dict: type[Mapping[str, Any]] = ..., preserve: bool = ...) -> None: ...
```

Comment:
Maybe this should be generic over the `_dict` type, so that we can get more accurate return types in `dump_sections()` and `get_empty_table()`.